src/MatrixClientPeg.ts

```diff
@@ -238,9 +238,10 @@ class MatrixClientPegClass implements IMatrixClientPeg {
             } catch (err) {
                 if (dbType === "indexeddb") {
                     logger.error("Error starting matrixclient store - falling back to memory store", err);
-                    this.matrixClient.store = new MemoryStore({
```

Comment:
Should the store field be made into a getter + private field maybe? To prevent this class of footgun

Reply:
Yeah that sounds better

Reply:
Or maybe just run the method using a `setter`